vaadin-login-flow-parent/vaadin-login-flow/src/main/java/com/vaadin/flow/component/login/AbstractLogin.java

```diff
@@ -222,4 +224,46 @@ public ForgotPasswordEvent(AbstractLogin source, boolean fromClient) {
     public void onEnabledStateChanged(boolean enabled) {
         getElement().setProperty(PROP_DISABLED, !enabled);
     }
+
+    /**
+     * Returns {@link LoginI18n} set earlier via {@link #setI18n(LoginI18n)}.
+     * <p>
+     * </p>
+     * Note that a copy of the original object is returned: changes done to the
+     * copy will not be reflected back until the object is set via
+     * {@link #setI18n(LoginI18n)}.
+     *
+     * @return currently set {@link LoginI18n} or null if none was set.
+     */
+    protected LoginI18n getI18n() {
+        final JsonValue json = (JsonValue) getElement().getPropertyRaw("i18n");
+        if (json == null || json instanceof JsonNull) {
+            return null;
+        }
+        return JsonSerializer.toObject(LoginI18n.class, json);
+    }
+
+    /**
+     * Shows given error message and sets {@link #setError(boolean)} to true.
+     *
+     * @param title
+     *            the {@link LoginI18n.ErrorMessage#getTitle() error message
+     *            title}, may be null.
+     * @param message
+     *            the {@link LoginI18n.ErrorMessage#getMessage() error message},
+     *            may be null.
+     */
+    public void setErrorMessage(String title, String message) {
+        var loginI18n = getI18n();
+        if (loginI18n == null) {
+            loginI18n = LoginI18n.createDefault();
+        }
+        if (loginI18n.getErrorMessage() == null) {
+            loginI18n.setErrorMessage(new LoginI18n.ErrorMessage());
+        }
+        loginI18n.getErrorMessage().setTitle(title);
+        loginI18n.getErrorMessage().setMessage(message);
+        setI18n(loginI18n);
+        setError(true);
```

Comment:
What is the motivation for calling `setError(true);`? In other components e.g. `TextField`, the `setErrorMessage` method only sets error text, but doesn't force it to be shown. IMO this method should behave the same.

Reply:
Good point! The motivation is that I'm calling this function overwhelmingly after the login failed:
```
try {
  login(username, password);
  navigateTo("");
} catch (LoginException ex) {
  setErrorMessage("Login failed", translate(ex.failureKey));
}
```
Therefore, I always want also the error to appear, and therefore I'd like the `setErrorMessage()` to call `setError(true)` since I assume this kind of omission will be the number 1 cause of problems.

That being said, `setErrorMessage()` indeed sounds similar to other setters, and calling `setError(true)` sounds like an unintended side-effect.

I'd still like to keep the call to `setError(true)`; how about if we rename the function to `showError()` or `showErrorMessage()` or such?

Reply:
Renamed to `showErrorMessage()`